tests/models/fastspeech2_conformer/test_tokenization_fastspeech2_conformer.py

```diff
@@ -0,0 +1,166 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the SpeechT5 tokenizers."""
+
+import unittest
+
+from transformers.models.fastspeech2_conformer import FastSpeech2ConformerTokenizer
+from transformers.testing_utils import require_g2p_en, slow
+
+from ...test_tokenization_common import TokenizerTesterMixin
+
+
+@require_g2p_en
+class FastSpeech2ConformerTokenizerTest(TokenizerTesterMixin, unittest.TestCase):
+    tokenizer_class = FastSpeech2ConformerTokenizer
+    test_rust_tokenizer = False
+
+    def setUp(self):
+        super().setUp()
+        tokenizer = FastSpeech2ConformerTokenizer.from_pretrained("connor-henderson/fastspeech2_conformer")
+        tokenizer.save_pretrained(self.tmpdirname)
+
+    def get_input_output_texts(self, tokenizer):
+        input_text = "this is a test"
+        output_text = "this is a test"
+        return input_text, output_text
+
+    # Custom `get_clean_sequence` since FastSpeech2ConformerTokenizer can't decode id -> string
+    def get_clean_sequence(self, tokenizer, with_prefix_space=False, **kwargs):  # max_length=20, min_length=5
+        input_text, output_text = self.get_input_output_texts(tokenizer)
+        ids = tokenizer.encode(output_text, add_special_tokens=False)
+        return output_text, ids
+
+    def test_convert_token_and_id(self):
+        """Test ``_convert_token_to_id`` and ``_convert_id_to_token``."""
+        token = "<unk>"
+        token_id = 1
+
+        self.assertEqual(self.get_tokenizer()._convert_token_to_id(token), token_id)
+        self.assertEqual(self.get_tokenizer()._convert_id_to_token(token_id), token)
+
+    def test_get_vocab(self):
+        vocab_keys = list(self.get_tokenizer().get_vocab().keys())
+
+        self.assertEqual(vocab_keys[0], "<blank>")
+        self.assertEqual(vocab_keys[1], "<unk>")
+        self.assertEqual(vocab_keys[-4], "UH0")
+        self.assertEqual(vocab_keys[-2], "..")
+        self.assertEqual(vocab_keys[-1], "<sos/eos>")
+        self.assertEqual(len(vocab_keys), 78)
+
+    def test_vocab_size(self):
+        self.assertEqual(self.get_tokenizer().vocab_size, 78)
+
+    def test_full_tokenizer(self):
+        tokenizer = self.get_tokenizer()
+
+        tokens = tokenizer.tokenize("This is a test")
+        ids = [9, 12, 6, 12, 11, 2, 4, 15, 6, 4, 77]
+        self.assertListEqual(tokens, ["DH", "IH1", "S", "IH1", "Z", "AH0", "T", "EH1", "S", "T", "<sos/eos>"])
+        self.assertListEqual(tokenizer.convert_tokens_to_ids(tokens), ids)
+        self.assertListEqual(tokenizer.convert_ids_to_tokens(ids), tokens)
+
+    @slow
+    def test_tokenizer_integration(self):
```

Comment:
Can we check the niche aspects of this tokenizer? E.g. do the properties in `_clean_text` work as expected?

Reply:
The `_clean_text` method is called within tokenize and so is implicitly tested in the `test_tokenizer_integration` test and its also the only somewhat niche method on this tokenizer. Did you have something in particular in mind?